### PR TITLE
feat: add `@Format` core annotation for Swagger and Json schema

### PR DESCRIPTION
**Pull request by rmcmk**
See: https://swagger.io/docs/specification/v3_0/data-models/data-types/#string-formats
See: https://json-schema.org/draft/2020-12/json-schema-validation#name-vocabularies-for-semantic-c

### CODE REVIEW

File: schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt [hunk 1]
```diff
@@ -1127,22 +1131,25 @@ class ReflectionParser_SwaggerGenerator_Tests : FunSpec({
...
                                 },
                                 "intValue": {
                                     "types": ["integer"],
                                     "format": "int32",
                                     "description": "Int field description",
                                     "default": "1111",
                                     "exampleSetFlag": true,
-                                    "example": "2222"
+                                    "example": "2222",
+                                    "format": "int32"
```
Comment (NecroKote) [inline schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt]: `format` specified twice. And it's in a lot of places

Comment (rmcmk) [reply schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt]: @NecroKote ~~3x per generator. Inline tests, referencing tests and referencing root tests.~~ Oops, I see, this test had an inferred format prior to the annotation being present. Resolved.

File: schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt [hunk 2]
```diff
@@ -1127,22 +1131,24 @@ class ReflectionParser_SwaggerGenerator_Tests : FunSpec({
...
+                                    "format": "text"
                                 },
                                 "intValue": {
                                     "types": ["integer"],
                                     "format": "int32",
                                     "description": "Int field description",
                                     "default": "1111",
                                     "exampleSetFlag": true,
-                                    "example": "2222"
+                                    "example": "2222",
```
Comment (SMILEY4) [inline schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt]:
```suggestion
                                    "example": "2222"
```
the json parser doesn't like trailing commas and throws an exception here

File: schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt [hunk 3]
```diff
@@ -1162,22 +1168,24 @@ class ReflectionParser_SwaggerGenerator_Tests : FunSpec({
...
```
Comment (SMILEY4) [inline schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt]:
```suggestion
                                    "example": "2222"
```
the json parser doesn't like trailing commas and throws an exception here

File: schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt [hunk 4]
```diff
@@ -1202,22 +1210,24 @@ class ReflectionParser_SwaggerGenerator_Tests : FunSpec({
...
+                                        "format": "text"
                                     },
                                     "intValue": {
                                         "types": ["integer"],
                                         "format": "int32",
                                         "description": "Int field description",
                                         "default": "1111",
                                         "exampleSetFlag": true,
-                                        "example": "2222"
+                                        "example": "2222",
```
Comment (SMILEY4) [inline schema-kenerator-test/src/test/kotlin/io/github/smiley4/schemakenerator/test/ReflectionParser_SwaggerGenerator_Tests.kt]:
```suggestion
                                        "example": "2222"
```
the json parser doesn't like trailing commas and throws an exception here

**Comment by SMILEY4** [comment]
I think we still need to add the extension methods for the steps in 
- https://github.com/SMILEY4/schema-kenerator/blob/develop/schema-kenerator-jsonschema/src/main/kotlin/io/github/smiley4/schemakenerator/jsonschema/steps.kt
- https://github.com/SMILEY4/schema-kenerator/blob/develop/schema-kenerator-swagger/src/main/kotlin/io/github/smiley4/schemakenerator/swagger/steps.kt
or rather add it to "handleCoreAnnotations()"

**Comment by rmcmk** [comment]
@SMILEY4 All fixed. Apologies for that, was moving fast. :smiley:

**Comment by SMILEY4** [comment]
no problem.   Thanks a lot for your pr :+1:

**Pull request merged**